vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1298,33 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private boolean hasSiblingWithSameType(Node child) throws RepositoryException {
+
+        Node parent = child.getParent();
+
+        try {
+            EffectiveNodeType ent = EffectiveNodeType.ofNode(parent);
+            String typeName = ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType()).get().getName();
+
+            NodeIterator iter = parent.getNodes();
+            while (iter.hasNext()) {
+                Node sibling = iter.nextNode();
+                if (!sibling.isSame(child)) {
+                    Optional<NodeDefinition> childDef = ent.getApplicableChildNodeDefinition(sibling.getName(),
+                            sibling.getPrimaryNodeType());
+                    try {
+                        if (typeName.equals(childDef.get().getName())) {
```

Comment:
Shouldn't we rather compare childDef.get().getDeclaringNodeType().getName() (with a null check?)

Reply:
I wonder whether comparing the type is correct at all. Consider a mandatory child node of type "nt:resource", and a folder that contains both an "nt:file" and an "nt:folder". Their node types are different, but they both satisfy the parent's requirement.

Reply:
Right, you need to take into account inheritance here.